### DISCUSSION TITLE
Ignoring field for

### DISCUSSION THREAD

**Discussion by y-nosa** [strawberry-graphql/strawberry#2596] [General]
Despite the fact there is no official support/integration of Sqlalchemy for Strawberry, I've been using strawberry with sqlalchemy for some time with success.
Strawberry works well with SqlAlchemy 2.0 when entities are modelled as data classes.
Recently I stumbled upon a specific case related to mapping the hierarchy of classes. Applying strawberry.type annotation breaks Sqlalchemy query generation when an additional field for polymorphic inference is mapped in Sqlalchemy.
```
class Base(DeclarativeBase, MappedAsDataclass):
    id: Mapped[int] = mapped_column(init=False, primary_key=True)

@strawberry.type
class Node1(Base):
    __tablename__ = 'nodes1'
    type: Mapped[str] = mapped_column( init=False, nullable=False)
    name: Mapped[str]


    __mapper_args__ = {
        'polymorphic_identity': 'node',
        'polymorphic_on': "type"
    }
```

With this additional mapping, sqlalchemy put additional semantics to the type field to determine the identity. 
Without digging into the details of the implementation for both Strawberry and SqlAlchemy, I thought that workaround would be to exclude this field from strawberry serialization. Nonetheless, cannot find any annotation for doing that.

Can you please point me out, if such an option is available?

**Comment by patrick91** [comment]
@y-nosa sorry for the late reply, have you seen [strawberry.Private](https://strawberry.rocks/docs/types/private#private-fields)?
